### PR TITLE
[F1 user signup] 

### PR DESCRIPTION
**Pull request by jamiecdavidson**
**Description**

1. SignUpPopup component created to replace sign up web page.
2. User can now successfully sign up the crazy bee
3. Change 3 ...

**Trello**

- Link: https://trello.com/c/IrBFkPwC/29-f1-user-signup-7

image
image

### CODE REVIEW

File: frontend/src/lib/validations.js [hunk 1]
```diff
@@ -28,4 +28,14 @@ const loginValidationSchema = Yup.object().shape({
   password: Yup.string().required('password is required'),
 })
 
-export { collectionValidationSchema, assetValidationSchema, loginValidationSchema }
+const signUpValidationSchema = Yup.object().shape({
+  email: Yup.string().email('must be a valid email').required('email is required'),
+  password: Yup.string().required('password is required'),
```
Comment (xuqun123) [inline frontend/src/lib/validations.js:33]: can we add validate rules for other fields as well?

**Comment by xuqun123** [comment]
Can we have some screenshots here?

One for the empty signup form, one for showing the form fields errors after hitting the `signup` button?

File: frontend/src/lib/validations.js (hunk 1 above)
Comment (jamiecdavidson) [reply frontend/src/lib/validations.js:33]: Thats right! I will add them in!

File: frontend/src/lib/validations.js [hunk 2]
```diff
@@ -28,4 +28,21 @@ const loginValidationSchema = Yup.object().shape({
...
+  username: Yup.string().required('username is required'),
+  dob: Yup.string(),
+  firstName: Yup.string().required('name is required'),
+  lastName: Yup.string(),
+  avatarUrl: Yup.string().required('profile picture is required'),
+  bannerImage: Yup.string(),
```
Comment (xuqun123) [inline frontend/src/lib/validations.js]: Wrong key, it should be called "bannerImageUrl"

**Comment by xuqun123** [comment]
Please change `bannerImage` to `bannerImageUrl` in everyplace from this PR.

Feel free to merge this PR after modifying that.

**Review by xuqun123 — APPROVED** [review]

File: frontend/src/lib/validations.js (hunk 2 above)
Comment (jamiecdavidson) [reply frontend/src/lib/validations.js]: amended

**Pull request merged**